pkg/gang_schedule/coscheduler/scheduler.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2021 The Alibaba Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package coscheduler
+
+import (
+	"context"
+	"sigs.k8s.io/scheduler-plugins/pkg/apis/scheduling/v1alpha1"
+
+	"github.com/alibaba/kubedl/apis"
+	"github.com/alibaba/kubedl/pkg/gang_schedule"
+	apiv1 "github.com/alibaba/kubedl/pkg/job_controller/api/v1"
+	"github.com/alibaba/kubedl/pkg/util/k8sutil"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func init() {
+	// Add to runtime scheme so that reflector of go-client will identify this CRD
+	// controlled by scheduler.
+	apis.AddToSchemes = append(apis.AddToSchemes, v1alpha1.AddToScheme)
+}
+
+func NewKubeCoscheduler(mgr controllerruntime.Manager) gang_schedule.GangScheduler {
+	return &kubeCoscheduler{client: mgr.GetClient()}
+}
+
+var _ gang_schedule.GangScheduler = &kubeCoscheduler{}
+
+type kubeCoscheduler struct {
+	client client.Client
+}
+
+func (kbs *kubeCoscheduler) Name() string {
+	return "kube-coscheduler"
```

Comment:
this name needs to be change to configurable, so that existing scheduler still works.
Can open a different issue for this

Reply:
+1, we should separate `Name()` interface into two methods: `Name()` and `SchedulerName()`, the previous one represents plugin-name and the latter one represents scheduler-name-in-cluster.

add a TODO issue